### DISCUSSION TITLE
Proposal: Introduce non-null-coalescing operator (opposite of null-coalescing operator)

### DISCUSSION THREAD

**Discussion by lauxjpn** [dotnet/csharplang#4136] [Language Ideas]
Currently there exists the widely popular `??` null-coalescing operator:

```c#
var myString = someString == null ? "my-default-string-value" : someString;
var myString = someString ?? "my-default-string-value";
```

However, there are also many instances, when you would need the opposite, for which I propose a `!?` non-null-coalescing operator:

```c#
var myString = someString != null ? $"This is {someString}" : null;
var myString = someString !? $"This is {someString}"; // <-- proposed non-null-coalescing operator syntax
```

**Comment by CyrusNajmabadi** [comment]
`someString !? $"This is {someString}"`  this is already a legal start of an expression.  `!` is the null suppression operator, and `?` is part of the ternary operator.
